azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+        return this;
+    }
+
+    /**
+     * @param annotations Key-Val containing non-identifying information
+     */
+    public AzKubernetesV1PodBuilder withPodAnnotations(Map<String, String> annotations) {
+        this.podMetaBuilder.withAnnotations(annotations);
+        return this;
+    }
+
+    /**
+     * @param name Flow-container/ application-container name
+     * @param image Docker image path in the image registry
+     * @param imagePullPolicy Docker image pull policy
+     * @param confVersion Version for the Azkaban configuration resource
+     * @return
+     */
+    public AzKubernetesV1PodBuilder addFlowContainer(String name, String image, String imagePullPolicy, String confVersion) {
+        V1EnvVar azConfVersion = new V1EnvVarBuilder()
+                .withName(AZ_CONF_VERSION_KEY)
+                .withValue(confVersion)
+                .build();
+        this.flowContainerBuilder
+                .withName(name)
+                .withImage(image)
+                .withImagePullPolicy(imagePullPolicy)
+                .withEnv(this.azClusterName, azConfVersion);
+        return this;
+    }
+
+    /**
+     * @param name JobType name to uniquely identify the init container names
+     * @param image Docker image path in the image registry
+     * @param imagePullPolicy Docker image pull policy
+     * @param initMountPath Path to be utilized by the jobType container image
+     * @param appMountPath Path mounted to flow-container/application-container corresponding to jobType
+     * @return
+     */
+    public AzKubernetesV1PodBuilder addJobType(String name, String image, String imagePullPolicy, String initMountPath, String appMountPath) {
+        String jobTypeVolumeName = JOBTYPE_VOLUME_PREFIX + name;
+        V1Volume jobTypeVolume = new V1VolumeBuilder()
+                .withName(jobTypeVolumeName)
+                .withNewEmptyDir()
+                .endEmptyDir()
+                .build();
+        V1EnvVar jobTypeMountPath = new V1EnvVarBuilder()
+                .withName(JOBTYPE_MOUNT_PATH_KEY)
+                .withValue(initMountPath)
+                .build();
+        V1VolumeMount initMountVolume = new V1VolumeMountBuilder()
+                .withName(jobTypeVolumeName)
+                .withMountPath(initMountPath)
+                .build();
+        V1VolumeMount appMountVolume = new V1VolumeMountBuilder()
+                .withName(jobTypeVolumeName)
+                .withMountPath(appMountPath)
+                .build();
+        V1Container initContainer = new V1ContainerBuilder()
+                .withName(JOBTYPE_INIT_PREFIX + name)
+                .addToEnv(this.azClusterName, jobTypeMountPath)
+                .withImagePullPolicy(imagePullPolicy)
+                .withImage(image)
+                .withVolumeMounts(initMountVolume)
+                .build();
+
+        this.appVolumes.add(jobTypeVolume);
+        this.appVolumeMounts.add(appMountVolume);
```

Comment:
Can you please add info and debug logs in this class?

Reply:
Done